src/main/java/uk/gov/ons/census/casesvc/service/CaseReceipter.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.ons.census.casesvc.service;
+
+import static uk.gov.ons.census.casesvc.utility.QuestionnaireTypeHelper.iscontinuationQuestionnaireTypes;
+
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.entity.UacQidLink;
+
+@Component
+public class CaseReceipter {
+  private final CaseService caseService;
+
+  public CaseReceipter(CaseService caseService) {
+    this.caseService = caseService;
+  }
+
+  public void handleReceipting(Case caze, UacQidLink uacQidLink) {
```

Comment:
is it reasonable to presume that the uacQidLink is always synonymous with the case being receipted? Why isn't the `active` flag being set to false on the `UacQidLink`?

Reply:
The idea was this method could be used for both new incoming receipts and receipting a case after a link comes in to an already receipted UacQidLink, in the second case we don't need to update the UacQidLink again (although is there any harm in setting the `active` flag false again?)

If this method is only going to handle the case specific stuff, I think it's name should reflect that at least.

Reply:
there are 5 lines of code in this function. Is this really achieving much?

Reply:
I think so, logic for receipting a case is getting more complex with incoming cards, doing this now puts that in one place where it'll be applied properly on new receipts and linking to already receipted QID's

Reply:
As Adam says we're purposefully moving all the logic to one place because we know that it's going to get more complex.  This will make it much easier to reuse & test.  

Naming wise I'll rename the function to 'ReceiptCase' ?